### PR TITLE
Fix some scatter plot issues

### PR DESCRIPTION
**Pull request by Illviljan**
Fix some issues with scatter plots:
* Always use markersize widths for scatter.
* Fix issue with `.values_unique` not returning the same values as `.values`
* Added more type hints to _Normalize, some rework had to be done to make mypy pass.

xref: #6778

### CODE REVIEW

**Comment by headtr1ck** [comment]
Unrelated to this PR but I noticed that typing of _Normalze was quite difficult because _unique_inverse can be a np.ndarray or DataArray, maybe you could clear that up a bit?

**Comment by Illviljan** [comment]
I've reworked _Normalize a little now. Hopefully a little easier to follow. Still not a fan of the optional None outputs, it adds too much extra code. Oh well, will save that for another rainy day.

File: xarray/plot/utils.py [hunk 1]
```diff
@@ -1475,17 +1474,23 @@ def _calc_widths(self, y: T | None) -> T | None:
...
-    def _indexes_centered(self, x: T) -> T | None:
+    @overload
+    def _indexes_centered(self, x: np.ndarray) -> np.ndarray:
+        ...
+
+    @overload
+    def _indexes_centered(self, x: DataArray) -> DataArray:
+        ...
+
+    def _indexes_centered(self, x: np.ndarray | DataArray) -> np.ndarray | DataArray:
```
Comment (headtr1ck) [inline xarray/plot/utils.py:1485]: Not sure why you didn't like the TypeVar here? But ok, the results are the same.

**Review by headtr1ck — APPROVED** [review]
You are right, the optional return make stuff a bit complicated. Not sure how to solve this though?
Maybe an extra class for the case of `data=None`?
Or it could be solved with a Generic class?
But I agree that this can be solved in another PR.

Thanks for addressing this!

File: xarray/plot/utils.py (hunk 1 above)
Comment (Illviljan) [reply xarray/plot/utils.py:1485]: I think mypy started complaining when I added typing to `.values`,  so I tried Union and finally overloads. This was one of the earliest changes though so maybe it works now when I changed ._unique_inverse ?

Comment (headtr1ck) [reply xarray/plot/utils.py:1485]: Maybe one of these instances where you need to use a TypeVar bound to a Union instead?

Comment (Illviljan) [reply xarray/plot/utils.py:1485]: Maybe, I'll tests this a bit more later once I find a new bug. This works good enough for now.

**Pull request merged**